src/lib/storage/disk_analyzer.rb

```diff
@@ -438,6 +415,36 @@ def remove_installation_disks(disks)
 
         disks.delete_if { |disk| @installation_disks.include?(disk.name) }
       end
+
+      # Find partitions from any of the candidate disks that have a given (set
+      # of) partition id(s).
+      #
+      # The result is a Hash in which each key is the name of a candidate disk
+      # and the value is an Array of DiskAnalyzer::Partition objects
+      # representing the matching partitions in that disk.
+      #
+      # @param ids [::Storage::ID, Array<::Storage::ID>]
+      # @return [Hash]
+      def partitions_with_id(ids)
+        pairs = candidate_disks.map do |disk_name|
+          partitions = disks.with(name: disk_name).partitions.with(id: ids).to_a
+          partitions.map! { |part| Partition.new(part.name, DiskSize.kiB(part.size_k)) }
+          [disk_name, partitions ]
+        end
+        Hash[pairs]
+      end
```

Comment:
I'm all for returning not just the device names; but why not use the existing Storage objects directly? Having
two different Partition classes doesn't help readability. And the same question comes up for \@root_device, etc...